### PR TITLE
[interp] Fix interp entry for methods with lots of arguments in llvmonly+interp mode.

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/mono/mono/issues/19801.

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/mini/interp/interp.c [hunk 1]
```diff
@@ -2819,17 +2817,22 @@ interp_create_method_pointer_llvmonly (MonoMethod *method, gboolean unbox, MonoE
...
 	 * a gsharedvt_in_sig wrapper.
+	 * We use a gsharedvt_in_sig wrapper instead of an interp_in wrapper, because they
+	 * are mostly the same, and they are already generated. The exception is the
+	 * wrappers for methods with more than 8 arguments, those are different.
 	 */
-	wrapper = mini_get_gsharedvt_in_sig_wrapper (sig);
+	if (sig->param_count > MAX_INTERP_ENTRY_ARGS)
+		wrapper = mini_get_interp_in_wrapper (sig);
+	else
+		wrapper = mini_get_gsharedvt_in_sig_wrapper (sig);
```
Comment (BrzVlad) [inline mono/mini/interp/interp.c:2827]: Why are we doing this on llvm as opposed to normal mode, where we always use the interp_in wrappers. They seem to be doing the same thing. Should we just use the interp_in_wrapper here for the sake of consistency ? Or is this meant to be an optimization that reduces the code size ?

File: mono/mini/aot-compiler.c [hunk 2]
```diff
@@ -8634,10 +8635,15 @@ add_gsharedvt_wrappers (MonoAotCompile *acfg, MonoMethodSignature *sig, gboolean
...
 
 #ifndef MONO_ARCH_HAVE_INTERP_ENTRY_TRAMPOLINE
-	if (interp_in) {
+	/*
+	 * Interpreter methods in llvmonly+interp mode are called using gsharedvt_in wrappers,
+	 * since we already generate those in llvmonly mode. But methods with a large
+	 * number of arguments need special processing (see interp_create_method_pointer_llvmonly),
+	 * which only interp_in wrappers do.
+	 */
+	if (interp_in || (gsharedvt_in && sig->param_count > MAX_INTERP_ENTRY_ARGS)) {
```
Comment (BrzVlad) [inline mono/mini/aot-compiler.c:8644]: Aren't we adding interp wrappers also on non interp aot images here ? While there aren't probably many methods satisfying this condition, looks rather ugly if so.

File: mono/mini/interp/interp.c (hunk 1 above)
Comment (vargaz) [reply mono/mini/interp/interp.c:2827]: We are already generating the gsharedvt_in wrappers, so we can reuse them for this as well.

File: mono/mini/aot-compiler.c (hunk 2 above)
Comment (vargaz) [reply mono/mini/aot-compiler.c:8644]: Will fix.

File: mono/mini/interp/interp.c (hunk 1 above)
Comment (vargaz) [reply mono/mini/interp/interp.c:2827]: Rewrote it to make it a bit clearer.

**Comment by jeromelaban** [comment]
Thanks for fixing this @vargaz !

**Pull request merged**